### DISCUSSION TITLE
Enabling HTTPS on WGDashboard

### DISCUSSION THREAD

**Discussion by ea1fsc** [WGDashboard/WGDashboard#504] [Q&A]
Hi all,
I would like to enable HTTPS in the WG Dashboard, as using it with HTTP is not secure for me because someone could sniff the traffic and discover my password for the Web GUI.
I have searched on the documentation and on Internet but I haven't found anything about how to do it.
I'm using a containerized way of WGDashboard. Is this an option that could be enabled someway or should I use reverse proxy servers or something similar to do it?

Thanks a lot :)

**Comment by DaanSelen** [comment]
Currently - WGDashboard does not yet support HTTPS, Donald is working on that.
If you use the container version you can setup: https://hub.docker.com/r/jc21/nginx-proxy-manager for HTTPS certificates! I do this myself.

Let me know if that helps!

**Reply by ea1fsc** [reply]
Hi Daan,
I have changed from the container version to the standard way of installing WGDashboard. I have tried installing nginx as a system package and then setting up the proxy, but I'm getting a 502: Bad Gateway.
I'm going to try with the container you send me and I'll tell you.

Thanks a lot! 
Juanchi EA1FSC

**Reply by DaanSelen** [reply]
Let me know if that works. I tried myself and it works great!

**Reply by asankahh** [reply]
![image](https://github.com/user-attachments/assets/8264711e-83e3-4d34-9445-e5c2d79e17ad)

I am using it and working totally fine. with docker

**Comment by waywit** [comment]
I have added my .key and .crt to src/gunicorn.conf.py:
```
certfile='/path/to/crt/file/file.crt'
keyfile='/path/to/key/file/file.key'
```
Now gunicorn is using these files for https... with the configured port.

BUT... every update of WGDashboard resets these setting ;-)

**Reply by ea1fsc** [reply]
Hi waywit,
Thanks for the solution, I will keep it in mind in case the proxy server doesn't work. Because I was looking for something that keeps in time (I'll probaly forget to set it up everytime I update WGDashboard :') ).

Juanchi EA1FSC

**Reply by NoahDar** [reply]
Hopefully Donald will see this and modify the install script to look for that.  Also, I've noticed in the script it got the letsencrypt check so makes me wonder if he is going to make use it for https.   I hope he does which will simplify things.

**Comment by airgapper** [comment]
Doesn’t proxing work ?   

Like Nginx:


server {
    listen 443 ssl;
    server_name your_domain_name.com;
    
    ssl_certificate /path/to/your/certificate.crt;
    ssl_certificate_key /path/to/your/certificate.key;
    
    location / {
        proxy_pass http://localhost:5000;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

Adam Lewis

**Reply by ea1fsc** [reply]
Hi Adam,
Thanks for your answer. I have tried this option, but when I try to access to WG Dashboard through the proxy server I'm getting a 502: Bad gateway.
I searched about it on Internet, but the solutions doesn't fix it. Do you know something about it?

Juanchi EA1SC

**Comment by jpizquierdo** [comment]
I have it working with my caddy reverse proxy using:
```
wireguard.example.com {
        handle {
                reverse_proxy 10.1.250.10:10086
        }
}

```

Caddy handles the certificate and I am able to access using TLS

**Comment by donaldzou** [comment]
Hi all, I'm creating a workflow to manage SSL certs to use on WGDashboard, basically there will be a `.ini` file, similar to `wg-dashboard.ini` storing the path for cert and pem file, and it won't be removed after every update. Will include it in the next release :)

**Comment by donaldzou** [comment]
Just created a PR: https://github.com/donaldzou/WGDashboard/pull/554

If you would like to test it out, please follow the guide in the PR :) Let me know if this works for you or not ;)
